identity/aziot-identityd/src/lib.rs

```diff
@@ -430,28 +488,78 @@ impl Server {
 				let csr = create_csr(
 					&subject,
 					&identity_public_key,
-					&identity_private_key)
+					&identity_private_key,
+					attributes)
 					.map_err(|err| Error::Internal(InternalError::CreateCertificate(Box::new(err))))?;
 				csr
 			};
 
 			let _new_cert = self.cert_client.create_cert(&identity_cert, &csr, None).await
-				.map_err(|err| Error::Internal(InternalError::CreateCertificate(Box::new(err))));
+				.map_err(|err| Error::Internal(InternalError::CreateCertificate(Box::new(err))))?;
 		}
 
 		Ok(())
 	}
 }
 
+fn deconstruct_local_opts(options: Option<settings::LocalIdOpts>, module_id: String) ->
+	(aziot_identity_common::LocalIdAttr, String, String) {
+	let defaults = (aziot_identity_common::LocalIdAttr::default(), module_id.clone(), module_id.clone());
+
+	options.map_or(defaults, |opts|
+		match opts {
+			settings::LocalIdOpts::X509 {attributes, cert_id, key_id,} => {
```

Comment:
Since module names are unique, can we not have the key id and cert id specified as options?

Reply:
Key id and cert id are specified as options. The module name is used for both key id and cert id if either is not specified.

Reply:
My suggestion was for not making key_id and cert_id configurable because I am not aware of any scenario where CS and KS are preloaded.

Reply:
Ah got it. Ok, I'll remove the options.